### ISSUE TITLE
Enhancement - on off schedule

### ISSUE THREAD

**Issue by erkr** [EPMatt/awesome-ha-blueprints#199]
### Blueprint name

on_off_schedule_state_persistence

### Description

Hi Matteo 
First thx for your nice blue print. I like it as it is. My feature request is to add sun set / sun rise (with some time offset) as on or off triggers. 
e.g lights on 15 minutes before sun-set, and lights off at 23:00h.
Regards Eric

### Additional Details

- [ ] I'd like to help developing this feature/improvement.

**Comment by SHxKM** [comment]
I came here to suggest this exactly! and I second the compliments on the schedule blueprint.

One enhancement I'd like to suggest for this blueprint:

> e.g lights on 15 minutes before sun-set, and lights off at 23:00h.

These can be dynamic attributes. `x` `minutes_before_sunset` for `light.on` and `y` `minutes_after_sunrise` for `light.off`.

**Comment by SHxKM** [comment]
I've actually started working on such blueprint but there's a huge blocker here because one cannot use variables in template triggers..

**Comment by erkr** [comment]
> I've actually started working on such blueprint but there's a huge blocker here because one cannot use variables in template triggers..

Maybe this blueprint can help: CyanAutomation/lights_on_at_sunset.yaml
The combination of these two blueprints would be great 😊

**Comment by SHxKM** [comment]
> > I've actually started working on such blueprint but there's a huge blocker here because one cannot use variables in template triggers..
> 
> Maybe this blueprint can help: CyanAutomation/lights_on_at_sunset.yaml The combination of these two blueprints would be great 😊

Sunset/sunrise is simple enough, but with an offset it becomes a whole different story because of inherent limitations currently imposed by HA :/. 

In fact, even writing an automation (not a blueprint) with state persistence similar to that in this repository requires some variable duplication.

**Comment by erkr** [comment]
Hi. That Blueprint supports an offset, not in time but a elevation delta. Works even better than a time offset as the sunset/rise speeds differ in summer vs winter

**Comment by SHxKM** [comment]
> Hi. That Blueprint supports an offset, not in time but a elevation delta. Works even better than a time offset as the sunset/rise speeds differ in summer vs winter

Yeah, I don’t disagree that sun-elevation is a more consistent measure between different seasons. But I was after a more naive and “simpler” blueprint, and that is unfortunately not really possible right now..
